### ISSUE TITLE
Avatar - Rejecting/Approving an outdated photo bug

### ISSUE THREAD

**Issue by Jambrose777** [thewca/worldcubeassociation.org#6424]
**Describe the bug**
When a WRT member accesses the approve avatars page, they see a list of pending avatars. The WRT member would then approve/reject the photos shown. If someone were to upload a new photo while the WRT member was on the pending avatar pages and the WRT member rejects the photo, the user will get an email stating that their avatar was rejected. The email to the user will include the most recently uploaded photo, while the WRT member would never have seen the new photo.

Note: If the WRT member were to approve a photo on the pending avatars' page after someone changed their photo uploaded, the correct photo that WRT approved will be on their profile and the most recent upload will be deleted.

# Rejection

**To Reproduce**
1. (User 1) Log in as a user.
2. (User 1) Click the top right profile dropdown toggle. Click on 'Edit Profile'.
3. (User 1) Click 'Avatar'. Click 'Choose File' and Upload Photo 1. Click the 'I confirm...' Checkbox. Click 'Save'.
4. (Admin) Log in as an admin (different session/window/device).
5. (Admin) Click the top right profile dropdown toggle. Click on 'Admin'
6. (Admin) Click on 'Approve pictures'. You Should see the pending avatar of Photo 1 for User 1.
7. (User 1) Repeat Step 3 for Photo 2: Click 'Avatar'. Click 'Choose File' and Upload Photo 2. Click the 'I confirm...' Checkbox. Click 'Save'.
8. (Admin) Click 'Reject' and fill in the reason. Click 'submit'.
9. (User 1) will receive an email about the rejection with Photo 2 in their email.

**Expected behavior**
Proposed Solution: When rejecting we can send the email for rejection with Photo 1 and **not** have Photo 2 Deleted. Then when the Approve Avatars page is reloaded after the WRT member clicks submit the page will show the Photo 2 which is still pending.

# Approval

**To Reproduce**
1. (User 1) Log in as a user.
2. (User 1) Click the top right profile dropdown toggle. Click on 'Edit Profile'.
3. (User 1) Click 'Avatar'. Click 'Choose File' and Upload Photo 1. Click the 'I confirm...' Checkbox. Click 'Save'.
4. (Admin) Log in as an admin (different session/window/device).
5. (Admin) Click the top right profile dropdown toggle. Click on 'Admin'
6. (Admin) Click on 'Approve pictures'. You Should see the pending avatar of Photo 1 for User 1.
7. (User 1) Repeat Step 3 for Photo 2: Click 'Avatar'. Click 'Choose File' and Upload Photo 2. Click the 'I confirm...' Checkbox. Click 'Save'.
8. (Admin) Click 'Approve'. Click 'submit'.
9. User 1 will now have Photo 1 as their avatar and Photo 2 will be deleted without any approval/denial of the photo.

**Expected behavior**
Proposed Solution: When approving Photo 1 we can **not** have Photo 2 Deleted. Then when the Approve Avatars page is reloaded after the WRT member clicks submit the page will show the Photo 2 which is still pending.

**Comment by gregorbg** [comment]
Another plain but simple solution: DOn't allow people to upload new avatar pictures while there's still a pending avatar request.
If necessary, add another "cancel pending avatar" button that needs to be pressed before people can upload a new avatar.

**Comment by dunkOnIT** [comment]
@gregorbg was this handled in the rework?

**Comment by gregorbg** [comment]
Not exactly, but it is immensely more trivial to fix now. Before, avatars used to be "crammed into" the users table, so that uploading another avatar would _override_ the pending one. Now, a new row is generated for every uploaded avatar, meaning that we'd have a much easier time discerning which is which.
